src/Chapter3.hs

```diff
@@ -970,7 +1189,30 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
-
+data DayOfWeek
+    = Monday 
+    | Tuesday 
+    | Wednesday 
+    | Thursday 
+    | Friday
+    | Saturday
+    | Sunday
+    deriving(Show, Enum, Bounded)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend (Saturday) = True
+isWeekend (Sunday) = True
+isWeekend (_) = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay (Sunday) = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` here:

```
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂